### PR TITLE
ENH: Add QAIC for GLM

### PR DESCRIPTION
**Pull request by kshedden**
Quasi-AIC for GLM with quasi-likelihood.

### CODE REVIEW

**Comment by kshedden** [comment]
@josef-pkt do you have any comments on this, especially from a design point of view?  Do we need a more general or standardized interface for AIC-like statistics?

**Comment by josef-pkt** [comment]
we need a method like `get_ic` or `get_infocrit`, or maybe just `infocrit`, that provides access to a variety of info criteria with options.
e.g. https://github.com/statsmodels/statsmodels/issues/1733#issuecomment-596129460

Adding an individual commonly used ic is ok, but overall we have too many of them and variations on them.
e.g. we have open issues whether auxiliary parameters like scale should be included in the count of parameters when computing ic. 

other ones like AICc are still missing, but AFAIR I have seen comments in the literature that the correction is only correct in gaussian case and not for many GLM families.

`qaic` might be a bit confusing as method because all other ic are properties
Is there a `qbic`? It looks to me that we could add the dispersion/scale correction to other ic as well.

related to misspecified loglike: TIC corresponds to robust cov_type and not just excess dispersion, #6516

File: statsmodels/genmod/generalized_linear_model.py [hunk 1]
```diff
@@ -1670,27 +1670,42 @@ def llnull(self):
...
-        """
-        Value of the loglikelihood function evalued at params.
-        See statsmodels.families.family for distribution-specific
-        loglikelihoods.
-        """
+    # Return the log-likelihood at the given scale, using the
+    # estimated scale if the provided scale is None.  In the Gaussian
+    # case with linear link, the concentrated log-likelihood is
+    # returned.
+    def _llf(self, scale=None):
```
Comment (josef-pkt) [inline statsmodels/genmod/generalized_linear_model.py:1677]: I think we could make this a public method `llf_scaled`
That would also be useful for a quasi-likelihood LR test with excess dispersion.

**Comment by josef-pkt** [comment]
additional related comments when I was working on model averaging https://github.com/statsmodels/statsmodels/issues/1802#issuecomment-560011153

**Comment by josef-pkt** [comment]
This needs unit test for cases with estimated scale like gaussian.
My guess would be that using self._llf(scale=1) is not correct in that case

Also, IIRC, then using `self._llf(scale=scale)` might add the correct fixed scale handling including overdispersion in Binomial and Poisson.

**Comment by kshedden** [comment]
@josef-pkt do you think this can be merged?

I added a warning for when QAIC might not make sense (as you pointed out).

**Comment by josef-pkt** [comment]
sorry I haven't gone back yet.

can you change the name to something like `infocrit` or `info_criteria`.

I want to reserve the `get_xxx` prefix now for more general post estimation results
get_prediction get_influence, get_diagnostic
(maybe change to get_margins)

**Comment by josef-pkt** [comment]
Looks good to me

Do you want to squash before merging?

either way is fine. 

I will add TIC when this is in master and option for allowing for extra params in the counting. 
Maybe `k_extra=0` or something like that, in negbin users might also want to add 1 for the dispersion parameter, or even 2 if for QMLE with both dispersion parameter and scale.

**Comment by josef-pkt** [comment]
maybe `dk_params` for delta k_params, similar to `ddof` in numpy

**Comment by josef-pkt** [comment]
something went wrong in the last rebase. It looks like you added many extra files that are not part of the repo

**Comment by kshedden** [comment]
I screwed this up when rebasing, so see #7283.

**Pull request closed without merging**